### DISCUSSION TITLE
plugins: How to pass data between steps?

### DISCUSSION THREAD

**Discussion by nitzano** [semantic-release/semantic-release#2180] [Q&A]
I'm writing a new plugin and trying to find a way to share data between steps 

For example (assume `test` is a valid option for the plugin):

```
async function verifyConditions(pluginConfig, context) {
  pluginConfig.test = "test1"
}

async function publish(pluginConfig, context) {
   console.log(publishConfig.test) // undefined, not "test1"
}
```

By inspecting https://github.com/semantic-release/github/blob/5f173f2ac2c7cf35bb28feb6a373d3d29c4cdeb9/index.js#L19
I thought settings data on `pluginConfig` should be available in further steps but  in my plugin  it
seems every step receive a "fresh" `pluginConfig` object (without any modifications from previous ones).

So my questions are:

1. Should the data set in [line 19](https://github.com/semantic-release/github/blob/5f173f2ac2c7cf35bb28feb6a373d3d29c4cdeb9/index.js#L19) be available in publish step as well? (line 31)?
2. What are the different ways to pass data between steps in a plugin?

**Comment by Kampfmoehre** [comment]
Is there any way now to pass data from one stop to later steps? If a prepare step of a plugin returns a value, is that somehow accessible in the context of the publish step?

**Comment by seebeen** [comment]
As far as I'm aware - sharing data between steps is not possible.

You could implement that with a function which uses a top level variable.
